src/Guesser/TypeGuesserChain.php

```diff
@@ -61,7 +62,7 @@ public function guessType($class, $property, ModelManagerInterface $modelManager
      * @param \Closure $closure The closure to execute. Accepts a guesser
      *                          as argument and should return a Guess instance
      *
-     * @return Guess The guess with the highest confidence
+     * @return TypeGuess The guess with the highest confidence
      */
     private function guess(\Closure $closure): Guess
```

Comment:
```suggestion
    private function guess(\Closure $closure): TypeGuess
```

Reply:
I ended up removing the method that I think it simplifies the code.